.github/workflows/build.yml

```diff
@@ -17,38 +17,62 @@ jobs:
   build:
     if: "!contains(github.event.head_commit.message, 'ci skip')"
     strategy:
+      fail-fast: false
       matrix:
+        container-image:
+          - "quay.io/fedora/fedora:rawhide"
+          - "quay.io/centos/centos:stream9"
+          - "registry.opensuse.org/opensuse/tumbleweed-dnf"
+          - "docker.io/library/ubuntu:22.04"
         compiler:
           - gcc
           - clang
         pam:
           - ON
           - OFF
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-latest
+
+    container:
+      image: ${{ matrix.container-image }}
+
     steps:
       - uses: actions/checkout@v3
-      - name: Dependencies
+      - name: Dependencies (rpm-type)
+        if: ${{ contains(matrix.container-image, 'fedora') || contains(matrix.container-image, 'centos') || contains(matrix.container-image, 'opensuse') }}
         run: |
           set -x
-          sudo sed -i '/^#\sdeb-src /s/^#//' "/etc/apt/sources.list"
-          sudo apt-get update -y
-          DEBIAN_FRONTEND=noninteractive sudo apt-get build-dep sddm -y
-          DEBIAN_FRONTEND=noninteractive sudo apt-get install clang clazy sudo qml-module-qttest -y
+          dnf --assumeyes install dnf-plugins-core
+          if [[ "${{ matrix.container-image }}" =~ "centos" ]]; then dnf --assumeyes config-manager --set-enabled crb && dnf --assumeyes install epel-release; fi
+          dnf --assumeyes builddep sddm
+          dnf --assumeyes install clang clazy
+      - name: Dependencies (deb-type)
+        if: ${{ contains(matrix.container-image, 'ubuntu') }}
+        run: |
+          set -x
+          sed -i '/^#\sdeb-src /s/^#//' "/etc/apt/sources.list"
+          apt-get update -y
+          DEBIAN_FRONTEND=noninteractive apt-get build-dep sddm -y
+          DEBIAN_FRONTEND=noninteractive apt-get install clang clazy qml-module-qttest -y
       - name: Build
         run: |
           set -x
           mkdir -p build
-          cd build
```

Comment:
Why are we changing from using `cd` to use weird arguments for `cmake` and `make`?

Reply:
It mimics how we actually build sddm downstream in distributions where those methods are used. I figure given that it looks more like how we build it and is slightly faster with less commands, it should be fine.

Reply:
It's considerably harder to read and it's not how anyone would do it on their system. I'd prefer if we keep it like it was.

I'd suggest you change how you do it in the packaging too. :)

Reply:
All distributions wrap CMake invocations through some interface. For RPM packages, we have one-liner macros:

```
%cmake
%cmake_build
%cmake_install
```

We're probably not going to change how those macros are implemented because it also makes it easy to twiddle to Ninja or something else without having to do anything too special. It also guarantees that they build out of tree.

(I could also technically drop the `mkdir -p build` command here too, since CMake will create it for me this way...)

But this isn't a hill I'll die on, so I'll roll it back...